### DISCUSSION TITLE
Component 和 Application 能否托管在公司内网中（比如私有源 npm）

### DISCUSSION THREAD

**Discussion by neurotoxinvx** [Serverless-Devs/Serverless-Devs#771] [Q&A]
因为代码暂时无法开源的缘故，我的 Component 和 Application 只能部署在公司内部，我目前尝试了 2 种方式：

- 方式一：download Application 模版到本地，然后通过 s init project --uri=local_app_template 的方式创建应用，但是这个方式使用起来较为麻烦，需要先下载，用户反馈不佳；

- 方式二：应用部署在 Gitlab 中，然后通过 s init git@sxxxxxxxxx 的方式创建。但是这个方式不会读取 publish.yaml 中配置的 Parameters，无法让用户输入 Prompt；

请问，还有其他的推荐方式嘛？或者，能否支持配置 npm 私有源 package 的方式拉取

**Comment by heimanba** [comment]
可以私下沟通此类需求 联系邮件 371510756@qq.com,或者加入钉钉群 33947367
